designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentStatus.scala

```diff
@@ -49,28 +47,84 @@ object DeploymentStatus extends Enum[DeploymentStatus] {
 
     def problemDescription: Option[String] = status match {
-      case _: NoAttributesDeploymentStatus             => None
+      case Finished(_)                                 => None
+      case DuringDeploy(_)                             => None
+      case Running(_, _)                               => None
+      case Canceled                                    => None
+      case DuringCancel                                => None
+      case Restarting                                  => None
       case ProblemDeploymentStatus(problemDescription) => Some(problemDescription)
     }
 
   }
 
+  implicit class ScenarioVersion(val status: DeploymentStatus) extends AnyVal {
+
+    def version: Option[VersionId] = status match {
+      case Finished(version)          => Some(version)
+      case DuringDeploy(version)      => Some(version)
+      case Running(version, _)        => Some(version)
+      case Canceled                   => None
+      case DuringCancel               => None
+      case Restarting                 => None
+      case ProblemDeploymentStatus(_) => None
+    }
+
+  }
+
+  implicit class StartedAt(val status: DeploymentStatus) extends AnyVal {
+
+    def startedAt: Option[Instant] = status match {
+      case Finished(_)                 => None
+      case DuringDeploy(_)             => None
+      case Running(version, startedAt) => Some(startedAt)
+      case Canceled                    => None
+      case DuringCancel                => None
+      case Restarting                  => None
+      case ProblemDeploymentStatus(_)  => None
+    }
+
+  }
+
   implicit class ToDeploymentStatusName(val status: DeploymentStatus) extends AnyVal {
 
     def name: DeploymentStatusName = status match {
-      case status: NoAttributesDeploymentStatus => DeploymentStatusName(status.entryName)
-      case ProblemDeploymentStatus(_)           => DeploymentStatusName.problemStatusName
+      case DuringDeploy(_)            => DeploymentStatusName.duringDeployStatusName
+      case Running(_, _)              => DeploymentStatusName.runningStatusName
+      case Finished(_)                => DeploymentStatusName.finishedStatusName
+      case Restarting                 => DeploymentStatusName.restartingStatusName
+      case DuringCancel               => DeploymentStatusName.duringCancelStatusName
+      case Canceled                   => DeploymentStatusName.canceledStatusName
+      case ProblemDeploymentStatus(_) => DeploymentStatusName.problemStatusName
     }
 
   }
 
-  def from(name: DeploymentStatusName, description: Option[String]): DeploymentStatus = {
+  def from(
+      name: DeploymentStatusName,
+      description: Option[String],
+      startedAt: Option[Instant],
+      modifiedAt: Instant,
+      version: Option[VersionId]
+  ): DeploymentStatus = {
     name match {
+      case DeploymentStatusName.duringDeployStatusName =>
+        DuringDeploy(version.getOrElse(VersionId(0)))
```

Comment:
Can we just throw an IllegalStateException in this situation and the ones below? If no, it is better to return -1, log a warning, and add a comment in the code.